pkg/virt-controller/watch/vm.go

```diff
@@ -991,18 +988,17 @@ func (c *VMController) handleVolumeUpdateRequest(vm *virtv1.VirtualMachine, vmi
 }
 
 func (c *VMController) addStartRequest(vm *virtv1.VirtualMachine) error {
-	addRequest := []virtv1.VirtualMachineStateChangeRequest{{Action: virtv1.StartRequest}}
-	req, err := json.Marshal(addRequest)
+	vm.Status.StateChangeRequests = append(vm.Status.StateChangeRequests, virtv1.VirtualMachineStateChangeRequest{Action: virtv1.StartRequest})
+	patchSet := patch.New()
+	patchSet.AddOption(patch.WithAdd("/status/stateChangeRequests", vm.Status.StateChangeRequests))
```

Comment:
Will `WithAdd()` work if there's already something under `/status/stateChangeRequests`?

Reply:
AFAIK the difference between add and replace is that add can create the path if it doesn't exist, while replace requires the path to already exist.

Reply:
Consider that this can be done inline:
```
patchBytes, err := patch.New(patch.WithAdd("/status/stateChangeRequests", vm.Status.StateChangeRequests)).GeneratePayload()
```